builtin/v9/migration/top.go

```diff
@@ -140,8 +150,35 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, newManifestCID
 		return cid.Undef, xerrors.Errorf("code cid for system actor not found in manifest")
 	}
 
+	miner9Cid, ok := newManifest.Get("storageminer")
+	if !ok {
+		return cid.Undef, xerrors.Errorf("code cid for miner actor not found in new manifest")
+	}
+
 	migrations[systemActor.Code] = systemActorMigrator{newSystemCodeCID, newManifest.Data}
 
+	// load market proposals
+	marketActor, ok, err := actorsIn.GetActor(builtin.StorageMarketActorAddr)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("failed to get market actor: %w", err)
+	}
+
+	if !ok {
+		return cid.Undef, xerrors.New("didn't find market actor")
+	}
+
+	var marketState market8.State
+	if err := store.Get(ctx, marketActor.Head, &marketState); err != nil {
+		return cid.Undef, xerrors.Errorf("failed to get system actor state: %w", err)
+	}
+
+	proposals, err := market8.AsDealProposalArray(adtStore, marketState.Proposals)
```

Comment:
I need someone to talk me through thread-safety here. I'm pretty sure this _isn't_ safe. 

If so, do we need to load the proposals AMT each time we migrate a miner [with deals in their pending precommits]? That wouldn't be too bad (ultimately not _that_ many such loads), but I'm wondering if people have better ideas.